### PR TITLE
Make the health-check path configurable.

### LINKED ISSUE #3577: [feature] Configurable health check route
Hello awesome community!

I've just a tiny feature request.

Transform health check hard coded route to configurable route.

Currently the route is: `/.well-known/apollo/server-health`.

Please just add this in config 🙏 

Thanks

## Annexes
- https://www.apollographql.com/docs/apollo-server/monitoring/health-checks/
- https://github.com/apollographql/apollo-server/blob/master/packages/apollo-server-express/src/ApolloServer.ts#L146

### PR DESCRIPTION
**Pull request by jutaz**
This is especially desirable for `apollo-server` package, which is OOTB pre-configured for quick usage, however, in few cases configuring the health-check endpoint is desired.

Certainly not set on the name of the property - better names welcome. I also think I've covered all integrations which support health checks out of the box, but LMK if I've missed any.

Closes #3577

### CODE REVIEW

**Comment by jutaz** [comment]
Tagging @abernix  for review (can't assign reviewers)

**Comment by glasser** [comment]
Hmm. One thing that's a challenge is that paths in the middlewares are sometimes relative to where the middleware is applied to your server and sometimes not.  But I guess just letting you swap out a hardcoded constant for a different one is reasonable.

I really wish that the health check was just a feature of `apollo-server` and people using a middleware integration just defined their own health check. It has very little to do with serving GraphQL and doesn't really fit with the rest of the project. Because of that, I'm tempted to say this new argument should *only* be an `apollo-server` thing, and if you want to customize your health check path with a different library then you can just define it in your own code.

**Comment by jutaz** [comment]
> I'm tempted to say this new argument should only be an apollo-server thing, and if you want to customize your health check path with a different library then you can just define it in your own code.

I guess this boils down to - are we viewing `apollo-server` as the only "batteries included" solution and other vendor-specific packages are "bring your own implementation"? I'm certainly impartial to the solution (personally only care about `apollo-server` which I'm using), but I wonder if there's a reason to still keep the abstraction for health checks in other packages.

TL;DR - I'll happily change this implementation to whatever maintainers think is best suited implementation, just LMK 👍

**Comment by glasser** [comment]
That's my perspective for sure. So yeah, let's just make this `apollo-server`-specific?

**Comment by jutaz** [comment]
@glasser Sorry for a slow response on this one - I've changed this PR so that only `apollo-server` and `apollo-server-express` (which is required to have this change due to the fact that `apollo-server` relies on it internally) would expose the `healthCheckPath`

**Comment by glasser** [comment]
Planning to come back to this once 3.0.0 is released (it's in final release candidate state now).

**Comment by gidich** [comment]
@glasser  - is this ready to merge in all checks are still green?

**Comment by glasser** [comment]
@gidich yep, got it on the queue (see label addition)

**Review by glasser — COMMENTED** [review]
(oops meant to send this yesterday)

**Comment by glasser** [comment]
I got a little carried away here and turned this into an opportunity to rewrite the docs and guidance around health checks.

**Comment by jutaz** [comment]
@glasser yup, makes sense - thanks for doing that!

File: packages/apollo-server-express/src/ApolloServer.ts [hunk 1]
```diff
@@ -22,6 +22,16 @@ export interface GetMiddlewareOptions {
   bodyParserConfig?: OptionsJson | boolean;
   onHealthCheck?: (req: express.Request) => Promise<any>;
   disableHealthCheck?: boolean;
+  // There's no real point to allowing you to customize the health check path in
+  // an Apollo Server web framework integration package. You're already using
+  // Express --- just define a health check yourself! This option only exists to
```
Comment (trevor-scheer) [inline packages/apollo-server-express/src/ApolloServer.ts:27]: Agree with the message but I think the way it's written might trivialize this a bit. Can we link out to some example or something?

**Review by trevor-scheer — APPROVED** [review]

**Review by trevor-scheer — APPROVED** [review]
